### PR TITLE
Bugfix/fix nano s enumeration bug

### PR DESCRIPTION
**Pull request by algoidan**
Goal was not able to communicate with the Ledger Nano S device.
The device has two HID occurrences on the OS and after the firmware upgrade, one of them became not responsive. That made goal to the hang.

+ fixing the interaction with the Ledger devices (S and X) on the Windows platform.

### CODE REVIEW

**Comment by nullun** [comment]
I just built and tested this and wanted to confirm it resolves the issues I was having here #2665 

Thank you @algoidan :+1:

File: daemon/kmd/wallet/driver/ledger_hid.go [hunk 1]
```diff
@@ -82,6 +83,13 @@ func (l *LedgerUSB) WritePackets(msg []byte) error {
 		if err != nil {
 			return err
 		}
+		// on Windows:
+		// The usb library adds one extra byte to the input passed to the USB device
+		// so the written bytes are larger than what we've send
+		// https://github.com/karalabe/hid/blob/9c14560f9ee858c43f40b5cd01392b167aacf4e8/hid_enabled.go#L167
+		if runtime.GOOS == "windows" && cc > 0 {
```
Comment (tsachiherman) [inline daemon/kmd/wallet/driver/ledger_hid.go:90]: consider refactoring so you would use ledger_windows.go for windows-specific logic.

File: daemon/kmd/wallet/driver/ledger_hid.go [hunk 2]
```diff
@@ -202,7 +210,13 @@ func LedgerEnumerate() ([]hid.DeviceInfo, error) {
 	}
 
 	var infos []hid.DeviceInfo
+	// The enumeration process is based on:
+	//  https://github.com/LedgerHQ/blue-loader-python/blob/master/ledgerblue/comm.py#L212
+	//  we search for the Ledger Vendor id and igonre devices that don't have specific usagepage or interface
 	for _, info := range hid.Enumerate(ledgerVendorID, 0) {
+		if info.UsagePage != 0xffa0 && info.Interface != 0 {
```
Comment (tsachiherman) [inline daemon/kmd/wallet/driver/ledger_hid.go:217]: maybe create some constant for this magic value ?

**Review by tsachiherman — APPROVED** [review]
looks good !

**Pull request merged**